core/src/main/java/nu/yona/server/analysis/entities/WeekActivity.java

```diff
@@ -104,22 +104,33 @@ public static WeekActivity createInstance(UserAnonymized userAnonymized, Goal go
 	public static WeekActivity createInstanceInactivity(UserAnonymized userAnonymized, Goal goal, ZonedDateTime startOfWeek)
 	{
 		WeekActivity result = createInstance(userAnonymized, goal, startOfWeek);
-		// add the days
-		// notice this doesn't take care of user time zone changes during the week
-		// so for consistency it is important that the batch script adding inactivity does so
-		for (int i = 0; i < 7; i++)
+		result.addInactivityDays();
+		return result;
+	}
+
+	public void addInactivityDays() 
+	{
+		// if the batch job has already run, skip
+		if(dayActivities.size() < 7)
```

Comment:
Can you pull this condition out of this method, to the caller? That way, ``addInactivityDays`` truly adds inactivity days. Otherwise rename this method into something like ``ensureFullWeek``. If you opt for the latter, switch the condition and make it return immediately when the week is already full. That way, you don't need to nest the body so deeply.